website/docs/docs/concepts/10-simple-schema.md

```diff
@@ -268,22 +268,70 @@ username: string | minLength=3 maxLength=20 pattern="^[a-zA-Z0-9_]+$"
 tags: "[]string" | uniqueItems=true minItems=1 maxItems=10 description="1-10 unique tags"
 ```
 
+:::warning Floating Point Precision
+When using `float` or `double` types in CEL expressions (particularly in `readyWhen` or `includeWhen` conditions), be aware of floating point precision issues that could cause unexpected behavior. Avoid comparing floating point values for equality in conditional logic. Prefer using `string`, `integer`, or `boolean` types whenever possible to avoid precision-related oscillations in resource state.
+:::
+
 ## Status Fields
 
 Status fields use CEL expressions to reference values from resources. kro
 automatically:
 
 - Infers the correct types from the expressions
-- Validates that referenced resources exist
+- Validates that referenced resources exist at ResourceGraphDefinition creation time
 - Updates values when the underlying resources change
+- Validates type compatibility using CEL's native type system
 
 ```yaml
 status:
   # Types are inferred from the referenced fields
-  availableReplicas: ${deployment.status.availableReplicas}
-  endpoint: ${service.status.loadBalancer.ingress[0].hostname}
+  availableReplicas: ${deployment.status.availableReplicas}  # integer
+  endpoint: ${service.status.loadBalancer.ingress[0].hostname}  # string
+  metadata: ${deployment.metadata}  # object
+```
+
+### Single vs Multi-Expression Fields
+
+Status fields can contain either a single CEL expression or multiple expressions concatenated together:
+
+**Single Expression Fields** can be any type:
+```yaml
+status:
+  replicas: ${deployment.status.replicas}  # integer
+  metadata: ${deployment.metadata}  # object
+  name: ${deployment.metadata.name}  # string
+  ready: ${deployment.status.conditions.exists(c, c.type == 'Available')}  # boolean
 ```
 
+**Multi-Expression Fields** (string templating) must contain only string expressions:
+```yaml
+status:
+  # ✓ Valid - all expressions return strings
+  endpoint: "https://${service.metadata.name}.${service.metadata.namespace}.svc.cluster.local"
```

Comment:
This is nice although CEL also has a way to do this AFAIK, `"%s:%d".format([self.spec.location, 7])` for example - maybe reads slightly less clearly though.

I assume we added special handling for this to make this work?